packages/dictionary/src/utils/typeUtils.ts

```diff
@@ -38,3 +76,42 @@ export const asArray = <T>(val: T | T[]): T[] => (Array.isArray(val) ? val : [va
  * const stringArray = combinedArray.filter(isDefined); // type is: Array<string>
  */
 export const isDefined = <T>(input: T | undefined) => input !== undefined;
+
+/**
+ * Determines if a variable is a number, with added restriction that it is Finite.
+ * This eliminates the values `NaN` and `Infinity`.
+ *
+ * Note: this is just a wrapper on `Number.isFinite` which is used by Lectern for identifying numbers in data value type checks.
+ * @param value
+ * @returns
+ */
+export const isNumber = (value: unknown): value is number => Number.isFinite(value);
+/**
+ * Determines if variable is of type number[], with added restriction that every element is Finite.
+ * @param value
+ * @returns
+ */
+export const isNumberArray = isArrayOf(isNumber);
+
+/**
+ * Determines if a variable is of type string[]
+ * @param value
+ * @returns
+ */
+export const isStringArray = isArrayOf((value: unknown) => typeof value === 'string');
+
+/**
+ * Determines if a variable is a number, with added restriction that it is an Integer.
+ *
+ * Note: This is a wrapper over `Number.isInteger` which is used by Lectern for identifying integers in data value type checks.
+ * @param value
+ * @returns
+ */
+export const isInteger = (value: unknown): value is number => Number.isInteger(value);
+
+/**
+ * Determines if a variables is of type number[], with add restriction that every element is an Integer.
+ * @param value
+ * @returns
+ */
+export const isIntegerArray = isArrayOf(isInteger);
```

Comment:
must be frank: though I get why this works, and it does well, I don't love how that reads out loud and that "smells faintly".